src/backend/base/langflow/components/vectorstores/astradb.py

```diff
@@ -502,34 +507,26 @@ def _initialize_collection_options(self):
             return []
 
     def update_build_config(self, build_config: dict, field_value: str, field_name: str | None = None):
-        if field_name == "embedding_choice":
-            if field_value == "Astra Vectorize":
-                build_config["embedding_model"]["show"] = False
-                build_config["embedding_model"]["required"] = False
-            else:
-                build_config["embedding_model"]["show"] = True
-                build_config["embedding_model"]["required"] = True
-
-            return build_config
-
-        if not self.token or not self.token.startswith("AstraCS:"):
-            build_config["database_name"]["info"] = "Add a Valid Token to Select a Database"
-        else:
-            build_config["database_name"]["info"] = "Select a Database from Astra DB"
-
         # Refresh the database name options
-        if field_name in ["token", "environment"] or not build_config["database_name"]["options"]:
-            # Reset the list of collections
-            build_config["collection_name"]["options"] = []
-            build_config["collection_name"]["options_metadata"] = []
-
+        if not os.getenv("LANGFLOW_HOST") and (
+            field_name in ["token", "environment"]
+            or (
+                field_name == "database_name"
+                and (
+                    not build_config["database_name"]["options"]
+                    or field_value != build_config["database_name"]["value"]
+                )
+            )
+            or not build_config["database_name"]["options"]
+        ):
```

Comment:
Should we create variables with descriptive names for these? It will be hard to maintain these conditions.

Reply:
That’s a really good point @ogabrielluiz , I’ll update it and also comment the build config changes better